api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,344 @@
+type HostedClusterInfrastructureSpec struct {
+
+	// InfraID is a globally unique identifier for the cluster. This identifier
+	// will be used to associate various cloud resources with the HostedCluster
+	// and its associated NodePools. If not specified the metadata.name for this
+	// resource is used. When specified, this value is used.
+	//
+	// +optional
+	// +immutable
+	InfraID string `json:"infraID,omitempty"`
+
+	// SubDomain is the identifier that is used between the *.apps and base-domain
+	// when building an ingress URL. This will also be the name of the cluster.
+	// If no value is provided, the infra-id is used
+	// When SubDomain = my and base-domain = company.com, the resulting ingress URL
+	// will be *.apps.my.company.com
+	//
+	// +optional
+	// +immutable
+	SubDomain string `json:"subDomain,omitempty"`
+
+	// Platform specifies the underlying infrastructure provider for the cluster
+	// and is used to configure platform specific behavior.
+	//
+	// +immutable
+	Platform PlatformInfraSpec `json:"platform"`
+
+	// baseDomain is the base domain of the cluster. All managed DNS records will
+	// be sub-domains of this base.
+	//
+	// For example, given the base domain `openshift.example.com`, an API server
+	// DNS record may be created for `cluster-api.openshift.example.com`.
+	//
+	// Once set, this field cannot be changed.
+	BaseDomain string `json:"baseDomain"`
+
+	// CloudProvider secret, contains the Cloud credenetial and Base Domain
+	// When not present, we expect all values to populated at create time
+	// This can be from the hypershift cli or via a kubectl create.
+	// +optional
+	CloudProvider corev1.LocalObjectReference `json:"cloudProvider,omitempty"`
+}
+
+// PlatformSpec specifies the underlying infrastructure provider for the cluster
+// and is used to configure platform specific behavior.
+type PlatformInfraSpec struct {
+	// Type is the type of infrastructure provider for the cluster.
+	//
+	// +kubebuilder:validation:Enum=AWS;Azure;PowerVS
+	// +immutable
+	Type PlatformType `json:"type"`
+
+	// AWS specifies configuration for clusters running on Amazon Web Services.
+	//
+	// +optional
+	// +immutable
+	AWS *AWSPlatformInfraSpec `json:"aws,omitempty"`
+
+	// Azure defines azure specific settings
+	Azure *AzurePlatformSpec `json:"azure,omitempty"`
+
+	// PowerVS specifies configuration for clusters running on IBMCloud Power VS Service.
+	// This field is immutable. Once set, It can't be changed.
+	//
+	// +optional
+	// +immutable
+	PowerVS *PowerVSPlatformSpec `json:"powervs,omitempty"`
+}
+
+// AWSPlatformSpec specifies configuration for clusters running on Amazon Web Services.
+type AWSPlatformInfraSpec struct {
+	// Region is the AWS region in which the cluster resides. This configures the
+	// OCP control plane cloud integrations, and is used by NodePool to resolve
+	// the correct boot AMI foHostedClusterInfrastructureConfigStatus a given release.
```

Comment:
>and is used by NodePool to resolve
// the correct boot AMI foHostedClusterInfrastructureConfigStatus a given release.

typo?

Do we really need this?

Reply:
fixed.